Added heron_control, changed heron_description to use with uuv_simulator

guystoppi opened the pull request.

guystoppi committed:
Added heron_control package

guystoppi committed:
Deleted unnecessary comments from CMakeLists.txt

guystoppi committed:
Updated robot_localization.yaml

Deleted unnecessary comments
Set frequency to 20Hz (the default update frequency of the um6 IMU).

guystoppi committed:
Updated package.xml description

guystoppi committed:
Initial commit of new heron_description files

guystoppi committed:
Updated version in package.xml

guystoppi committed:
Added navsat/vel to robot_localization.yaml

guystoppi committed:
Switched UUV Sensor plugins to Gazebo plugins

guystoppi committed:
Re-added copyright notice in heron.urdf.xacro

guystoppi committed:
Additional updates so Gazebo sensor plugins would work with namespaces

guystoppi committed:
Changed simulated GPS topic to match the actual topic

guystoppi committed:
Fixed robot_localization to expect TwistWithCovarianceStamped for navsat/vel

guystoppi committed:
Removed uuv_msgs dependencies (not actually needed)

guystoppi committed:
Put covariance-adding script into heron_control pkg

guystoppi committed:
Made changes to allow the actual Heron robot to use heron_description

guystoppi committed:
Added envvars to change sensor locations and attached GPS info to separate sensor frame

guystoppi committed:
Renamed navsat/vel topic to allow Vec3->Twist translation

guystoppi committed:
Removed NWU->ENU swapping because it's only needed for simulation

Commit:
Fixed name+email in copyright notices

Commit:
Made global covariance accessible from navsat/fix callback

Commit:
Further implementing namespaces

tonybaltovski commented on heron_description/CHANGELOG.rst:
```diff
@@ -2,6 +2,10 @@
 Changelog for package heron_description
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+0.4.0 (2018-06-26)
```
Remove this, it get automatically generated when you use `catkin_generate_changelog`

tonybaltovski commented on heron_description/package.xml:
```diff
@@ -1,7 +1,7 @@
 <package>
   <name>heron_description</name>
-  <version>0.3.0</version>
+  <version>0.4.0</version>
```
Revert, this is done using `catkin_prepare_release`

tonybaltovski commented on heron_description/urdf/sensors.xacro:
```diff
@@ -0,0 +1,80 @@
+     <headingDrift>0.005</headingDrift>
+     <headingGaussianNoise>0.005</headingGaussianNoise>
+   </plugin>
+ </gazebo>
```
Spacing is off.

tonybaltovski commented on heron_description/CMakeLists.txt:
```diff
@@ -1,9 +1,13 @@
 
-find_package(catkin REQUIRED COMPONENTS roslaunch)
+find_package(catkin REQUIRED COMPONENTS
+    heron_msgs
```
Are `heron_msgs` needed?  I don't see them used any where.

guystoppi replied:
Should I put "roslaunch" back in?

dniewinski commented on heron_description/urdf/base.xacro:
```diff
@@ -0,0 +1,137 @@
+  <!-- Collision geometry mesh, usually in STL format (it is recommended to keep
+  this geometry as simple as possible to improve the performance the physics engine
+  regarding the computation of collision forces) -->
+  <xacro:property name="collision_mesh_file" value="file://$(find heron_description)/meshes/heron_base.stl"/>
```
Does the simulator use the collision mesh to calculate buoyancy?  If so, simplifying this mesh may go a long way for efficiency

Commit:
Undoing package number changes

Commit:
Fixed spacing using Atom auto-indent

Commit:
Added simpler collision mesh

Commit:
Removed heron_msgs dependency and added roslaunch back

Commit:
Fixing namespaces to allow for empty namespace

Commit:
Added functionality of GPS datum via env var

tonybaltovski commented:
@dniewinski any remaining concerns?

david0429 commented:
None on my end

dniewinski commented:
LGTM

tonybaltovski commented:
@Nirzvi1 can you rebase?

Commit:
Edited package.xml and moved heron_controller node from heron_control

Commit:
Removed initial slash to allow for namespace prefixing

Commit:
Added heron_control package (#1)

* Added heron_control package

* Deleted unnecessary comments from CMakeLists.txt

* Updated robot_localization.yaml

Deleted unnecessary comments
Set frequency to 20Hz (the default update frequency of the um6 IMU).

* Edited package.xml and moved heron_controller node from heron_control

* Updated package.xml description

Commit:
Allowed use of odometry yaw for simulation

Commit:
Fixing mistake from rebasing

guystoppi committed:
Merge branch 'kinetic-devel' into kinetic-devel
